desc/compute/utils.py

```diff
@@ -1325,6 +1326,190 @@ def body(i, mins):
     return grid.expand(mins, surface_label)
 
 
+def bounce_integral(eq, lambdas, rho=None, alpha=None, resolution=20):
+    """Returns a method to compute the bounce integral of any quantity.
+
+    The bounce integral is defined as F_ℓ(λ) = ∫ f(ℓ) / √(1 − λ |B|) dℓ, where
+        dℓ parameterizes the distance along the field line,
+        λ is a constant proportional to the magnetic moment over energy,
+        |B| is the norm of the magnetic field,
+        f(ℓ) is the quantity to integrate along the field line,
+        and the endpoints of the integration are at the bounce points.
+    For a particle with fixed λ, bounce points are defined to be the location
+    on the field line such that the particle's velocity parallel to the
+    magnetic field is zero, i.e. λ |B| = 1.
+
+    Parameters
+    ----------
+    eq : Equilibrium
+        Equilibrium on which the bounce integral is defined.
+    lambdas : ndarray
+        λ values to evaluate the bounce integral at.
+    rho : ndarray
+        Unique flux surface label coordinates.
+    alpha : ndarray
+        Unique field line label coordinates over a constant rho surface.
+    resolution : int
+        Number of quadrature points used to compute the bounce integral.
+
+    Returns
+    -------
+    bi : callable
+        This callable method computes the bounce integral F_ℓ(λ) for every
+        specified field line ℓ (constant rho and alpha), for every λ value in
+        ``lambdas``.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        bi = bounce_integral(eq, lambdas)
+        F = bi(name)
+
+    """
+    if rho is None:
+        rho = jnp.linspace(0, 1, 10)
+    if alpha is None:
+        alpha = jnp.linspace(0, 2 * jnp.pi, 20)
+
+    # Use Gauss-Chebyshev quadrature as the integrand blows up at integration boundary.
+    x, w = chebgauss(deg=resolution)
```

Comment:
The weights and integration points are straightforward and have analytical expresssions. We can just write a function for that or make chebgauss jitable in by redefining it is desc.backend? Low-priority suggestion.

Reply:
Yes if chebguass isn't jitable (I assumed it was since there were analytic expressions, but I should check) we can define an alias for it that is jitable in `desc.backend`.